include/sparrow/iterator.hpp

```diff
@@ -48,7 +48,7 @@ namespace sparrow
                 {
                 }
 
-                Reference* operator->() const
+                Reference* operator->()
                 {
                     return std::addressof(m_reference);
                 }
```

Comment:
Nope, because it will makes `m_reference` a `const Reference` in this method, making it impossible to bind its address of type `const Reference*` to the return type `Reference*`. For const iterator, the constness is already "encoded" in the `Reference` type.